src/vs/workbench/api/common/extHostExtensionService.ts

```diff
@@ -824,20 +825,26 @@ export abstract class AbstractExtHostExtensionService extends Disposable impleme
 			throw err;
 		};
 
+		const getResolver = async (remoteAuthority: string) => {
+			logInfo(`activating resolver for ${remoteAuthority}...`);
+			const { resolver, authorityPrefix } = await this._activateAndGetResolver(remoteAuthority);
+			if (!resolver) {
+				logError(`no resolver for ${authorityPrefix}`);
+				throw new RemoteAuthorityResolverError(`No remote extension installed to resolve ${authorityPrefix}.`, RemoteAuthorityResolverErrorCode.NoResolverFound);
+			}
+			return { resolver, authorityPrefix, remoteAuthority };
+		};
+
 		const chain = remoteAuthorityChain.split(/@|%40/g).reverse();
 		logInfo(`activating remote resolvers ${chain.join(' -> ')}`);
 
 		let resolvers;
 		try {
-			resolvers = await Promise.all(chain.map(async remoteAuthority => {
-				logInfo(`activating resolver...`);
-				const { resolver, authorityPrefix } = await this._activateAndGetResolver(remoteAuthority);
-				if (!resolver) {
-					logError(`no resolver`);
-					throw new RemoteAuthorityResolverError(`No remote extension installed to resolve ${authorityPrefix}.`, RemoteAuthorityResolverErrorCode.NoResolverFound);
-				}
-				return { resolver, authorityPrefix, remoteAuthority };
-			}));
+			resolvers = await Promise.all(chain.map(getResolver)).catch(async (e: Error) => {
+				if (e instanceof RemoteAuthorityResolverError) { throw e; }
```

Comment:
Did you mean to make this instead

```suggestion
				if (!RemoteAuthorityResolverError.isNoResolverFound(e)) { throw e; }
```

Reply:
I think here we should do fallback only when error is ```Error(`Not an authority that can be resolved!`)``` - and it is not subclass of `RemoteAuthorityResolverError`

Reply:
Yes, I agree. The way the logic is implemented, it will only fall back to `getResolver(remoteAuthorityChain)` if `e` is _not_ a subclass of RemoteAuthorityResolverError

Reply:
Oh, sorry, I misunderstood.

I think we should introduce a new RemoteAuthoriyResolverErrorCode for an "invalid authority" to avoid having to catch generic errors here, which could lead to incorrectly falling back (e.g. if an error is thrown from a remote extension' activate method)

Reply:
I don't really understand why there are two different classes for RemoteAuthorityResolverError:
https://github.com/microsoft/vscode/blob/main/src/vs/platform/remote/common/remoteAuthorityResolver.ts#L97
https://github.com/microsoft/vscode/blob/main/src/vs/workbench/api/common/extHostTypes.ts#L536

But import in this file points to the second one, which does not have `.isNoResolverFound` methods.